Configurations.md

```diff
@@ -803,6 +803,140 @@ By default this option is set as a percentage of [`max_width`](#max_width) provi
 
 See also [`max_width`](#max_width) and [`width_heuristics`](#width_heuristics)
 
+## `fn_generics_space`
+
+Leave spaces around function generics.
+
+- **Default value**: `"Never"`
+- **Possible values**: `"OnlyBefore"`, `"OnlyAfter"`, `"BeforeAndAfter"`, `"Never"`
+- **Stable**: No (tracking issue: [#3564](https://github.com/rust-lang/rustfmt/issues/3564))
+
+#### `"Never"` (default):
+
+```rust
+fn lorem() {
+    // body
+}
+
+fn lorem(ipsum: usize) {
+    // body
+}
+
+fn lorem<T>(ipsum: T)
+where
+    T: Add + Sub + Mul + Div,
+{
+    // body
+}
+```
+
+#### `"OnlyBefore"`:
+
+```rust
+fn lorem() {
+    // body
+}
+
+fn lorem(ipsum: usize) {
+    // body
+}
+
+fn lorem <T>(ipsum: T)
+where
+    T: Add + Sub + Mul + Div,
+{
+    // body
+}
+```
+
+#### `"OnlyAfter"`:
+
+```rust
+fn lorem() {
+    // body
+}
+
+fn lorem(ipsum: usize) {
+    // body
+}
+
+fn lorem<T> (ipsum: T)
+where
+    T: Add + Sub + Mul + Div,
+{
+    // body
+}
+```
+
+#### `"BeforeAndAfter"`:
+
+```rust
+fn lorem() {
+    // body
+}
+
+fn lorem(ipsum: usize) {
+    // body
+}
+
+fn lorem <T> (ipsum: T)
+where
+    T: Add + Sub + Mul + Div,
+{
+    // body
+}
+```
+
+See also [`fn_no_generics_space`](#fn_no_generics_space).
+
+## `fn_no_generics_space`
```

Comment:
Same bikeshed here. What would you think about `fn_sig_name_paren_space` or `fn_name_paren_space` and then in the descriptions for the two config options (and with inline comments in example) we can re-emphasize that this one's only used in signatures that have no generics and vice versa?

Reply:
I added comments inline, but I prefer `fn_no_generics_spacing`. I wouldn't expect `fn_name_paren_space` to only apply without generics. It's also not very clear that it's the complement of `fn_generics_spacing`

Reply:
> I wouldn't expect fn_name_paren_space to only apply without generics. It's also not very clear that it's the complement of fn_generics_spacing

I always try to think about folks using config options they aren't terribly familiar with and doing so either via cli override or from their editor in a rustfmt config file vs. always being on the config documentation page, and that's why I generally prefer the names to be as explicit and self-explaining as possible.

My original thought was that by itself, the name just hints at some spacing for a function signature that lacks generics vs. explicitly whether it'd inject a space between the ident of the function name and the opening paren. I was trying to think of something would more clearly indicate _what_ the config option did and not just _when_ it would do something.

I do take your other point though, so let's see how the broader discussion in this PR plays out and then maybe we can iterate a bit more on the name to see if we can address both concerns :+1:

Reply:
> that's why I generally prefer the names to be as explicit and self-explaining as possible.

I agree. This is very helpful

> let's see how the broader discussion in this PR plays out and then maybe we can iterate a bit more on the name to see if we can address both concerns

:thumbsup: